homeassistant/components/notify/rest.py

```diff
@@ -0,0 +1,79 @@
+"""
+homeassistant.components.notify.rest
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+REST platform for notify component.
+
+For more details about this platform, please refer to the documentation at
+https://home-assistant.io/components/notify.rest/
+"""
+import logging
+import requests
+
+from homeassistant.helpers import validate_config
+from homeassistant.components.notify import (
+    DOMAIN, ATTR_TITLE, ATTR_TARGET, BaseNotificationService)
+
+_LOGGER = logging.getLogger(__name__)
+
+DEFAULT_METHOD = 'GET'
+DEFAULT_MESSAGE_PARAM_NAME = 'message'
+DEFAULT_TITLE_PARAM_NAME = None
+DEFAULT_TARGET_PARAM_NAME = None
+
+
+def get_service(hass, config):
+    """ Get the REST notification service. """
+
+    if not validate_config({DOMAIN: config},
+                           {DOMAIN: ['resource', ]},
+                           _LOGGER):
+        return None
+
+    method = config.get('method', DEFAULT_METHOD)
+    message_param_name = config.get('message_param_name',
+                                    DEFAULT_MESSAGE_PARAM_NAME)
+    title_param_name = config.get('title_param_name',
+                                  DEFAULT_TITLE_PARAM_NAME)
+    target_param_name = config.get('target_param_name',
+                                   DEFAULT_TARGET_PARAM_NAME)
+
+    return RestNotificationService(config['resource'], method,
+                                   message_param_name, title_param_name,
+                                   target_param_name)
+
+
+# pylint: disable=too-few-public-methods, too-many-arguments
+class RestNotificationService(BaseNotificationService):
+    """ Implements notification service for REST. """
+
+    def __init__(self, resource, method, message_param_name,
+                 title_param_name, target_param_name):
+        self._resource = resource
+        self._method = method
```

Comment:
If you store the method uppercase you will not have to convert it on every invocation.